source/aws-ruby-sdk/s3_helper.rb

```diff
@@ -0,0 +1,19 @@
+require 'aws-sdk-s3'
+
+module IdentityIdpFunctions
+  class S3Helper
+    def download(url)
+      uri = URI.parse(url)
+      bucket = uri.host.gsub('.amazonaws.com', '')
```

Comment:
I don't think this will extract the bucket name correctly, which can be in the format of either...

- https://[bucket].s3.[region].amazonaws.com (e.g. https://my-bucket.s3.us-west-2.amazonaws.com)
- https://s3.[region].amazonaws.com/[bucket] (e.g. https://s3.us-west-2.amazonaws.com/my-bucket)

The URLs generated by the Ruby AWS S3 SDK default configuration for presigning appear to use the path format. I've verified this in Doug's personal sandbox.

Example test case: https://github.com/18F/identity-idp/blob/191436d/spec/helpers/aws_s3_helper_spec.rb#L45-L46

Reference:

- https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html#access-bucket-intro
- https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html#presigned_url-instance_method (`virtual_host` option)

Reply:
Ok thank for confirming, I just picked up this code and moved it, I'll make sure to update it

Reply:
Handling both cases in https://github.com/18F/identity-idp-functions/pull/40, in case we ever switch

Reply:
> in case we ever switch

Which is an interesting point in itself, since...

- Apparently the path scheme is being deprecated ([source](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/))
- ...but the host scheme will yield HTTP (not HTTPS) URLs for presigning ([source](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Presigner.html#presigned_url-instance_method))

Neither of which is ideal 🤷

Reply:
😭